WordPress/gutenberg
# Components: Lighten borders to gray-600

Closes #39425

## What?

Updates the border color of our control components (e.g. InputControl, SelectControl, etc) so it is a shade lighter (gray-600 instead of 700).

## Why?

To reflect the updated design spec.

## How?

With the theming work (#44116) in mind, we'll use a semantically named intermediary variable (something that includes the word "border") instead of using the `$gray-600` variable directly.

## Testing Instructions

1. `npm run storybook:dev`
2. You can smoke test some control components with borders, e.g. TextControl, UnitControl.
3. This PR isn't going to cause any serious issues, because the difference between gray-600 and 700 is subtle. But if you want to see what exactly is being affected, you can rewrite the border color variables so its value is `magenta` or something, and the changes will be more obvious in Storybook.

## Screenshots or screencast 

| Before | After |
|--------|-------|
|![InputControl border, before](https://user-images.githubusercontent.com/555336/205104119-c5fc15fc-6a0f-46db-bfe9-5394e69a3a93.png)|![InputControl border, after](https://user-images.githubusercontent.com/555336/205104127-41ea192f-db8a-4d4a-89ae-0490766ac8f3.png)|
